### ISSUE TITLE
optical map info stays 'unmapped'

### ISSUE THREAD

**Issue by noschenk** [tanghaibao/jcvi#102]
Hi there,

I am having a problem with 'path'. I want to scaffold my NGS contigs with information from an optical map (Bionano) and a genetic map. I created a .bed file  [by hand](https://github.com/noschenk/useful_material/blob/master/raw_xmap2allmaps.py) from the Bionano .xmap output.
Merging with `mergebed` resulted in the file 'merged.bed':

```
Peex113Ctg00004	145	146	geneticmap-L1:60.332313	Peex113Ctg00004:146
Peex113Ctg00011	19611	316215	opticalmap-SuperScaffold33563:19611.0	28.09	+	Peex113Ctg00011:19612
Peex113Ctg00011	345625	813900	opticalmap-SuperScaffold15046:8187.0	58.21	+	Peex113Ctg00011:345626
Peex113Ctg00043	283	284	geneticmap-L1:82.301015	Peex113Ctg00043:284
Peex113Ctg00043	5744	456884	opticalmap-SuperScaffold50:6031.0	50.10	+	Peex113Ctg00043:5745
Peex113Ctg00043	524934	869428	opticalmap-SuperScaffold13968:3705.0	38.33	+	Peex113Ctg00043:524935
```

After running `path`, unfortunately, I find only the information of "geneticmap" in the resulting pdfs, nothing about opticalmaps.
![l3_allmaps](https://user-images.githubusercontent.com/32162523/42571874-8c2c3b20-8518-11e8-834b-669897bad350.png)
 All (or most) of the optical mapping SuperScaffolds appear in the file "unmapped". Either with the comment _#Partially deleted in new_ or _# Deleted in new_. (I write partly because opticalmap.bed has 1904 lines, while the file "unmapped" has 3450/2=1725)

Here the first three lines of "unmapped":
```
#Partially deleted in new
Peex113Ctg00011	19611	316215	opticalmap-SuperScaffold33563:19611.0	28.09	+	Peex113Ctg00011:19612
#Deleted in new
Peex113Ctg00011	345625	813900	opticalmap-SuperScaffold15046:8187.0	58.21	+	Peex113Ctg00011:345626
#Deleted in new
Peex113Ctg00011	1008645	1177090	opticalmap-SuperScaffold11:178212.0	15.74	-	Peex113Ctg00011:1008646
```

First, I thought maybe my contigs are chimeric, so I followed the instruction to split chimeric contigs. However, no chimeric contigs were found for optical mapping and for the genetic map, first 13 chimeric contigs were found, but after `gaps` and `refine`, the resulting .fasta file was exactly the same as my input. That means I can suggest there has no correction be done, right?

Do you have any inputs for me? Am I searching at the wrong edge?
I appreciate all inputs, 
Best, noelle

**Comment by tanghaibao** [comment]
@noschenk 
I'll answer the question about splitting chimeric contigs first. If you don't have gaps (NNNN) in the FASTA sequence, the split won't work (the tool was developed at a time when splitting scaffolds were a key goal, not contigs).

For optical mapping markers missing entirely in the output, this is quite puzzling to me. Can you verify that the `*.lifted.bed` contained only genetic markers and none of the optical mapping markers? This may be an edge bug that you have run into. Also I assumed that you could run the test-data okay?

Haibao

**Comment by noschenk** [comment]
Dear Haibao, 
thanks for responding that quickly!
Thanks for the answer, the absence of splitting totally makes sense now.

The test data of medicago runned well. Here I show you 3 lines from the .lifted.bed file. In the pdf output, all three maps are there.
```
Scaffold0108	463043	468191	Chickpea-1:376.10976	Scaffold0108:463044
Scaffold0108	470358	471009	Chickpea-1:376.07674	Scaffold0108:470359
...
Scaffold0138	332362	332363	GeneticMap-lg1:366.925	Scaffold0138:332363
...
```

Here is the head of the `*lifted.bed` file output when I reran the analysis with my data (after restart): 
```
Peex113Ctg00046	24282	480548	opticalmap-SuperScaffold46:456444.0	41.91	-	Peex113Ctg00046:24283
Peex113Ctg00046	494211	678752	opticalmap-SuperScaffold6359:227691.0	19.64	-	Peex113Ctg00046:494212
Peex113Ctg00054	3585	165214	opticalmap-SuperScaffold5554:626.0	20.55	+	Peex113Ctg00054:3586
Peex113Ctg00095	9805	532626	opticalmap-SuperScaffold5595:9805.0	45.99	+	Peex113Ctg00095:9806
Peex113Ctg00133	1178	1179	geneticmap-L1:119.959487	Peex113Ctg00133:1179
Peex113Ctg00539	392542	630082	opticalmap-SuperScaffold11060:7896.0	27.97	+	Peex113Ctg00539:392543
Peex113Ctg00554	38301	434233	opticalmap-SuperScaffold15082:422398.0	35.54	-	Peex113Ctg00554:38302
Peex113Ctg00629	8145	594856	opticalmap-SuperScaffold1679:8145.0	75.06	+	Peex113Ctg00629:8146
Peex113Ctg00773	51759	353347	opticalmap-SuperScaffold36299:8646.0	40.38	+	Peex113Ctg00773:51760
Peex113Ctg00825	489	490	geneticmap-L7:10.705364	Peex113Ctg00825:490
```
There are 179 lines saying 'opticalmap' in the 4th column.

I was watching the output of `path` and observed that there were 108 lines saying " Skipping opticalmap-SuperScaffoldX", of total 1904 optical map scaffolds. 
Additionally, I found many lines saying `AMBIGUOUS Peex113Ctg18488 {('opticalmap-SuperScaffold15175',): 1, ('opticalmap-SuperScaffold14938',): 1}`, that are 549 lines, of those around 500 unique ones. 

The past day I have been revising my optical mapping data output because I thought maybe the problem is my data, which could be messy? I know that the genome of the sister species has been quite tricky to assemble and scaffold. I will go on checking there as well.

Best, Noelle

**Comment by noschenk** [comment]
Hi,
I wanted to give a little update. In the meantime, I have been shuffling the data around as following: I used a new version of the genome, one that has been created with the optical map data I have previously been working with. The genetic map has been adapted to this new genome version. Thus with `ALLMAPS`, I only merged one genetic map and the new genome version. With this updated dataset, `ALLMAPS` runned as expected.

I am not sure what caused the problem exactly, but the new version of the genome contains some splitted contig from the old version. Apparently, there were some misassemblies.

Thanks again a lot for your help, 
Best, Noelle
